pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/LaunchSparkDataIngestionJobCommand.java

```diff
@@ -0,0 +1,368 @@
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.JavaVersion;
+import org.apache.commons.lang3.SystemUtils;
+import org.apache.pinot.spi.auth.AuthProvider;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.apache.pinot.spi.filesystem.PinotFS;
+import org.apache.pinot.spi.filesystem.PinotFSFactory;
+import org.apache.pinot.spi.ingestion.batch.IngestionJobLauncher;
+import org.apache.pinot.spi.ingestion.batch.spec.PinotFSSpec;
+import org.apache.pinot.spi.ingestion.batch.spec.SegmentGenerationJobSpec;
+import org.apache.pinot.spi.plugin.PluginManager;
+import org.apache.pinot.spi.utils.GroovyTemplateUtils;
+import org.apache.pinot.tools.Command;
+import org.apache.spark.launcher.SparkLauncher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+
+/**
+ * Class to implement LaunchDataIngestionJob command.
+ *
+ */
+@CommandLine.Command(name = "LaunchSparkDataIngestionJob")
+public class LaunchSparkDataIngestionJobCommand extends AbstractBaseAdminCommand implements Command {
+  private static final Logger LOGGER = LoggerFactory.getLogger(LaunchSparkDataIngestionJobCommand.class);
+  public static final String MAIN_CLASS = "org.apache.pinot.tools.admin.command.LaunchDataIngestionJobCommand";
+  public static final String SPARK_HOME = "SPARK_HOME";
+  public static final String BASEDIR = "basedir";
+  public static final String LOCAL_FILE_PREFIX = "local://";
+  public static final String PINOT_MAIN_JAR_PREFIX = "pinot-all";
+
+  @CommandLine.Option(names = {"-help", "-h", "--h", "--help"}, required = false, help = true, description = "Print "
+      + "this message.")
+  private boolean _help = false;
+  @CommandLine.Option(names = {"-jobSpecFile", "-jobSpec"}, required = true, description = "Ingestion job spec file")
+  private String _jobSpecFile;
+  @CommandLine.Option(names = {"-values"}, required = false, arity = "1..*", description = "Context values set to the"
+      + " job spec template")
+  private List<String> _values;
+  @CommandLine.Option(names = {"-propertyFile"}, required = false, description = "A property file contains context "
+      + "values to set the job spec template")
+  private String _propertyFile;
+  @CommandLine.Option(names = {"-pluginsToLoad"}, required = false, arity = "1..*", split = ":", description = "List "
+      + "of plugin name separated by : to load at runtime. e.g. pinto-s3:pinot-parquet")
+  private List<String> _pluginsToLoad;
+
+  // Kafka plugins need to be excluded as they contain scala dependencies which cause
+  // NoSuchMethodErrors with runtime spark.
+  // It is also fine to exclude Kafka plugins as they are not going to be used in batch ingestion in any case
+  @CommandLine.Option(names = {"-pluginsToExclude"}, defaultValue = "pinot-kafka-0.9:pinot-kafka-2.0", required =
```

Comment:
i wonder if we already hav a property file and a jobspec file. do we still want to support these commandline options

Reply:
Yes, that can be done as well. My initial thought process when writing this was to not use ingestion spec at all in the command code and only use it inside the spark job. Reason being it puts the limitation of always providing ingestion spec as a local file path and not S3.

However, I did ended up using the spec because otherwise we can't load the PinotFS classes and find appropriate plugin jars in S3, GCS etc.

Reply:
I will take this up as a seperate PR